src/components/lineup/Lineup.tsx

```diff
@@ -0,0 +1,295 @@
+const MAX_TILES_COUNT = 1000
+
+// The max number of loading tiles to display if count prop passes
+const MAX_COUNT_LOADING_TILES = 18
+
+// The inital multiplier for number of tracks to fetch on lineup load
+// multiplied by the number of tracks that fit the browser height
+export const INITIAL_LOAD_TRACKS_MULTIPLIER = 1.75
+export const INITIAL_PLAYLISTS_MULTIPLER = 1
+
+// A multiplier for the number of tiles to fill a page to be
+// loaded in on each call (after the intial call)
+const TRACKS_AHEAD_MULTIPLIER = 0.75
+
+// Threshold for how far away from the bottom (of the list) the user has to be
+// before fetching more tracks as a percentage of the list height
+const LOAD_MORE_THRESHOLD = 0.5
+
+// The minimum inital multiplier for tracks to fetch on lineup load
+// use so that multiple lineups on the same page can switch w/out a reload
+const MINIMUM_INITIAL_LOAD_TRACKS_MULTIPLIER = 1
+
+// tile height + margin
+const totalTileHeight = {
+  main: 152 + 16,
+  section: 124 + 16,
+  condensed: 124 + 8,
+  playlist: 350
+}
+
+// Helpr to calculate an item count based on the Lineup variant and a multiplier
+export const getItemCount = (
+  variant: LineupVariant,
+  multiplier: number | (() => number)
+) =>
+  Math.ceil(
+    (Dimensions.get('window').height / totalTileHeight[variant]) *
+      (typeof multiplier === 'function' ? multiplier() : multiplier)
+  )
+
+// Calculate minimum, initial, and loadMore itemCounts
+const useItemCounts = (variant: LineupVariant) =>
+  useMemo(
+    () => ({
+      minimum: getItemCount(
+        variant === LineupVariant.PLAYLIST
+          ? LineupVariant.PLAYLIST
+          : LineupVariant.MAIN,
+        MINIMUM_INITIAL_LOAD_TRACKS_MULTIPLIER
+      ),
+      initial: getItemCount(variant, () =>
+        variant === LineupVariant.PLAYLIST
+          ? INITIAL_PLAYLISTS_MULTIPLER
+          : INITIAL_LOAD_TRACKS_MULTIPLIER
+      ),
+      loadMore: getItemCount(variant, TRACKS_AHEAD_MULTIPLIER)
+    }),
+    [variant]
+  )
+
+const styles = StyleSheet.create({
+  item: {
+    padding: 12,
+    paddingBottom: 0
+  }
+})
+
+/** `Lineup` encapsulates the logic for displaying a list of items such as Tracks (e.g. prefetching items
+ * displaying loading states, etc).
+ */
+export const Lineup = ({
+  actions,
+  count,
+  delineate,
+  isTrending,
+  leadingElementId,
+  limit,
+  lineup,
+  loadMore,
+  header,
+  playTrack,
+  pauseTrack,
+  rankIconCount = 0,
+  refresh,
+  refreshing,
+  showLeadingElementArtistPick = true,
+  variant
+}: LineupProps) => {
+  const dispatchWeb = useDispatchWeb()
+
+  const playing = useSelector(getPlaying)
+  const playingUid = useSelector(getPlayingUid)
+  const itemCounts = useItemCounts(variant)
+
+  // Item count based on the current page
+  const pageItemCount = useMemo(() => {
+    return itemCounts.initial + (lineup.page - 1) * itemCounts.loadMore
+  }, [itemCounts, lineup])
+
+  // Either the provided count or a default
+  const countOrDefault = useMemo(
```

Comment:
just curious if this needs to be a memo? i believe nums can be recomputed without triggering rerender?

Reply:
It doesn't, just a habit of putting potentially expensive computations in a memo so that they aren't recomputed on every render. But this one isn't expensive so I can remove the memo